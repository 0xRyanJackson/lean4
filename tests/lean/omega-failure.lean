/-!
Testing omega's failure message
-/

example : 0 < 0 := by omega

example (x : Nat) : x < 0 := by omega

example (x : Nat) (y : Int) : x < y := by omega

example (x y : Int) : 5 < x ∧ x < 10 → y > 0 := by omega

theorem sizeOf_snd_lt_sizeOf_list {α : Type u} {β : Type v} [SizeOf α] [SizeOf β] {x : α × β} {xs : List (α × β)} :
  x ∈ xs → sizeOf x.snd < 1 + sizeOf xs
:= by
  intro h
  have := List.sizeOf_lt_of_mem h
  have : sizeOf x = 1 + sizeOf x.1 + sizeOf x.2 := rfl
  omega
