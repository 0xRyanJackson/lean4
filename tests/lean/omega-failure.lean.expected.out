omega-failure.lean:5:22-5:27: error: omega could not prove the goal:
it is trivially false
omega-failure.lean:7:32-7:37: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  x₁ ≥ 0
where
  x₁ := ↑x
omega-failure.lean:9:42-9:47: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  x₁ - x₂ ≥ 0
  x₁ ≥ 0
where
  x₁ := ↑x
  x₂ := y
omega-failure.lean:11:51-11:56: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  x₂ ≤ 0
  6 ≤ x₁ ≤ 9
where
  x₁ := x
  x₂ := y
omega-failure.lean:19:2-19:7: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  x₅ - x₆ ≥ 1
  x₆ ≥ 0
  x₅ ≥ 0
  x₁ - x₂ ≥ 1
  x₄ ≥ 0
  x₂ ≥ 0
  x₁ ≥ 0
  x₄ + x₅ ≥ -1
where
  x₁ := ↑(sizeOf xs)
  x₂ := ↑(sizeOf x)
  x₄ := ↑(sizeOf x.fst)
  x₅ := ↑(sizeOf x.snd)
  x₆ := ↑(sizeOf xs)
